cmake/mcuboot.cmake

```diff
@@ -0,0 +1,182 @@
+# Included by mcuboot/zephyr/CMakeLists.txt
+
+if(CONFIG_BOOTLOADER_MCUBOOT)
+  function(sign to_sign_hex output_prefix offset sign_depends signed_hex_out)
+    set(op ${output_prefix})
+    set(signed_hex ${op}_signed.hex)
+    set(${signed_hex_out} ${signed_hex} PARENT_SCOPE)
+    set(to_sign_bin ${op}_to_sign.bin)
+    set(update_bin ${op}_update.bin)
+    set(moved_test_update_hex ${op}_moved_test_update.hex)
+    set(test_update_hex ${op}_test_update.hex)
+
+    add_custom_command(
+      OUTPUT
+      ${update_bin}            # Signed binary of input hex.
+      ${signed_hex}            # Signed hex of input hex.
+      ${test_update_hex}       # Signed hex with IMAGE_MAGIC
+      ${moved_test_update_hex} # Signed hex with IMAGE_MAGIC located at secondary slot
+
+      COMMAND
+      # Create signed hex file from input hex file.
+      # This does not have the IMAGE_MAGIC at the end. So for this hex file
+      # to be applied by mcuboot, the application is required to write the
+      # IMAGE_MAGIC into the image trailer.
+      ${sign_cmd}
+      ${to_sign_hex}
+      ${signed_hex}
+
+      COMMAND
+      # Create binary version of the input hex file, this is done so that we
+      # can create a signed binary file which will be transferred in OTA
+      # updates.
+      ${CMAKE_OBJCOPY}
+      --input-target=ihex
+      --output-target=binary
+      ${to_sign_hex}
+      ${to_sign_bin}
+
+      COMMAND
+      # Sign the binary version of the input hex file.
+      ${sign_cmd}
+      ${to_sign_bin}
+      ${update_bin}
+
+      COMMAND
+      # Create signed hex file from input hex file *with* IMAGE_MAGIC.
+      # As this includes the IMAGE_MAGIC in its image trailer, it will be
+      # swapped in by mcuboot without any invocation from the app. Note,
+      # however, that this this hex file is located in the same address space
+      # as the input hex file, so in order for it to work as a test update,
+      # it needs to be moved.
+      ${sign_cmd}
+      --pad # Adds IMAGE_MAGIC to end of slot.
+      ${to_sign_hex}
+      ${test_update_hex}
+
+      COMMAND
+      # Create version of test update which is located at the secondary slot.
+      # Hence, if a programmer is given this hex file, it will flash it
+      # to the secondary slot, and upon reboot mcuboot will swap in the
+      # contents of the hex file.
+      ${CMAKE_OBJCOPY}
+      --input-target=ihex
+      --output-target=ihex
+      --change-address ${offset}
+      ${test_update_hex}
+      ${moved_test_update_hex}
+
+      DEPENDS
+      ${sign_depends}
+      )
+  endfunction()
+
+  if (CONFIG_MCUBOOT_BUILD_S1_VARIANT)
+    # Inject this configuration from parent image to mcuboot.
+    set(conf_path "${ZEPHYR_BASE}/../nrf/subsys/bootloader/image/build_s1.conf")
```

Comment:
```suggestion
     set(conf_path "${ZEPHYR_NRF_MODULE_DIR}/subsys/bootloader/image/build_s1.conf")
```

Reply:
I was planning on leaving the contents of the file as close to the original one being moved, but I agree with you :)